examples/rln-js/index.js

```diff
@@ -0,0 +1,493 @@
+      const reversedArray = idCommitment.slice().reverse();
+      const pubkey = ethers.utils.hexlify(reversedArray).toString();
+
+      const price = await rlnContract.MEMBERSHIP_DEPOSIT();
+
+      const signer = provider.getSigner();
+      const rlnContractWithSigner = rlnContract.connect(signer);
+
+      const txResponse = await rlnContractWithSigner.register(pubkey, {
+        value: price,
+      });
+      console.log("Transaction broadcasted:", txResponse);
+
+      const txReceipt = await txResponse.wait();
+
+      console.log("Transaction receipt", txReceipt);
+
+      event = txReceipt.events[0];
+    }
+
+    // Update membershipId
+    membershipId = event.args.index.toNumber();
+
+    console.log(
+      "Obtained index for current membership credentials",
+      membershipId
+    );
+
+    ui.setRlnStatus("Successfully registered.");
+    ui.setMembershipInfo(membershipId, membershipKey);
+    ui.enableDialButton();
+  });
+
+  return {
+    encoder,
+    rlnInstance,
+  };
+}
+
+async function initWaku(ui, rln) {
+  ui.setWakuStatus("Creating Waku node.");
+  const node = await createLightNode();
+  ui.setWakuStatus("Starting Waku node.");
+
+  await node.start();
+  ui.setWakuStatus("Waku node started.");
+
+  const verifyMessageAndRender = (message) => {
+    if (message.proofState === "verifying...") {
+      try {
+        console.log("Verifying proof without roots");
+        console.time("proof_verify_timer");
+        const res = message.verifyNoRoot();
+        console.timeEnd("proof_verify_timer");
+        console.log("proof verified without roots", res);
+        if (res === undefined) {
+          message.proofState = "no proof attached";
+        } else if (res) {
+          message.proofState = "verified.";
+        } else {
+          message.proofState = "invalid!";
+        }
+      } catch (e) {
+        message.proofState = "Error encountered, check console";
+        console.error("Error verifying proof:", e);
+      }
+
+      ui.renderMessage(message);
+      console.log("Verifying proof with roots", message.verify());
+    }
+  };
+
+  const onFilterMessage = (wakuMessage) => {
+    const { timestamp, nick, text } = ProtoChatMessage.decode(
+      wakuMessage.payload
+    );
+
+    const time = new Date();
+    time.setTime(Number(timestamp) * 1000);
+
+    if (wakuMessage.rateLimitProof) {
+      console.log("Proof received:", wakuMessage.rateLimitProof);
+    }
+
+    wakuMessage.proofState = !!wakuMessage.rateLimitProof
+      ? "verifying..."
+      : "no proof attached";
+
+    wakuMessage.msg = `
+            (${nick})
+            <strong>${utils.bytesToUtf8(text)}</strong>
+            <i>[${time.toISOString()}]</i>
+        `;
+
+    verifyMessageAndRender(wakuMessage);
+  };
+
+  ui.onDial(async (ma) => {
+    ui.setWakuStatus("Dialing peer.");
+
+    // TODO(@weboko): move this fix into Waku.dial
+    const multiaddr = MultiformatsMultiaddr.multiaddr(ma);
```

Comment:
Noticed this issue in other examples I am migrating right now. 
Will create a PR on the side of js-waku to cover it.
Approach will be to add a parsing from string to `mutliaddr` structure to prevent any friction with libp2p. In new version they do not parse string and just throw an error.

Reply:
This one should resolve the issue. https://github.com/waku-org/js-waku/pull/1306